tests/pass/shims/fs.rs

```diff
@@ -279,6 +276,12 @@ fn test_directory() {
             .collect::<BTreeMap<_, _>>()
     );
     // Deleting the directory should fail, since it is not empty.
+    #[cfg(any(target_os = "solaris", target_os = "illumos"))]
+    // Solaris/Illumos `rmdir` call set errno to EEXIST if directory contains
+    // other entries than `.` and `..`.
+    // https://docs.oracle.com/cd/E86824_01/html/E54765/rmdir-2.html
+    assert_eq!(ErrorKind::AlreadyExists, remove_dir(&dir_path).unwrap_err().kind());
```

Comment:
This assertion seems to fail on illumos now.

Reply:
Locally it passes tough, dunno if this is due to native vs non native run.

Reply:
"Locally" meaning on an Illumos system? If you are passing through host errors, and host errors differ between OSes, then Miri will have to adjust the errors.

This likely also means that on your system, the tests will fail if you run them with `--target x86_64-unknown-linux-gnu`?

Reply:
> Does anyone really rely on the _specific_ errors being emitted?

The standard library does/has relied on specific error codes being returned by standard library functions, and has panicked or hit a debug assertion when an unexpected error code is produced.

I thought "pass through to the host" was something we explicitly avoided?

Reply:
> I thought "pass through to the host" was something we explicitly avoided?

For IO functions it is unclear what else we could practically do. We already pass through the actual file system of the host (i.e., this anyway requires disabling isolation), also passing through the errors codes doesn't make a big difference any more at that point.

> The standard library does/has relied on specific error codes being returned by standard library functions, and has panicked or hit a debug assertion when an unexpected error code is produced.

Yes, but AFAIK this usually only checks that the error is some code allowed by POSIX -- whereas here we are apparently talking about POSIX allowing different codes, and different Unixes indeed using different codes.

Reply:
Oh dear, I see what's up here. Solarish is returning EEXIST for an attempt to remove a non-empty directory, because the POSIX description is "File Exists". But then std translates this error code to `ErrorKind::AlreadyExists`, adding the "already" which at least in this specific case, is definitely wrong. But when we display an io::Error, we punt to `strerror` which should only say "File exists" so we loop back to reasonable behavior.

So I'm okay with some non-portability here. Is not the first time the ErrorKind translation has been rough.

Reply:
Okay, so let's p;ass through the host error, and let's make the test accept both `AlreadyExists` and `NotEmpty`.